### DISCUSSION TITLE
nodejs client

### DISCUSSION THREAD

**Discussion by kilfour** [ravendb/ravendb#19148] [Q&A]
Hi,

I have previously used the C# client extensively and with much joy, but for a new project I'm looking into the nodeJs client, and I just can't to seem to get the simplest examples to work the way I expect them to. 
My first issue is that session.Store does not seem to set the id of the object even if I try to retrieve it with session.advanced.getDocumentId.
I'm just using the code provided on the RavenDb documentation page. Any thoughts on what is going on ?

**Comment by ayende** [comment]
What does your entity looks like?
We are calling:

https://github.com/ravendb/ravendb-nodejs-client/blob/2bd5fc2d64151b5ec5cb5466ad37f6f8274921bb/src/Documents/Identity/GenerateEntityIdOnTheClient.ts#L69

Which end up defaulting to `id` if you didn't specify anything else.

**Comment by kilfour** [comment]
I use an entity definition from one of  the examples :
```
class TodoItem {
    constructor(opts) {
        opts = opts || {};
        this.id = `null;`
        this.content = `opts.content;`
        this.createdAt = opts.createdAt;
        this.isChecked = opts.isChecked || false;
    }
}
```
but after :
```
const item = new TodoItem({
    content: "Test",
    createdAt: new Date(),
    isChecked: false
});

const session = store.openSession();
session
    .store(item)
    .then(session.saveChanges())
    .then(console.log(session.advanced.getDocumentId(item)), reason => console.log(reason));
```

the console logs 'null'

**Reply by ayende** [reply]
Did you _intend_ to write this:

```
this.id = `null;`
```

You are setting the ID to the string "null;" ?

**Reply by ayende** [reply]
I wrote this standalone code:

```
const ravendb = require('ravendb');

class TodoItem {
    constructor(opts) {
        opts = opts || {};
        this.id = null;
        this.content = opts.content;
        this.createdAt = opts.createdAt;
        this.isChecked = opts.isChecked || false;
    }
}

(async function () {

    const store = new ravendb.DocumentStore('http://live-test.ravendb.net', 'todo-app');

    store.initialize();

    const session = store.openSession();
    var item = new TodoItem({ content: 'hello there' });
    await session.store(item);
    console.log("before", item.id);
    await session.saveChanges();
    console.log("after", item.id);

    store.dispose();
}())
```

The output is:

```
before TodoItems/1-A
after TodoItems/1-A
```
